### ISSUE TITLE
File links does not work properly

### ISSUE THREAD

**Issue by Diafwl** [secure-77/Perlite#55]
**Describe the bug**
links does not work properly for files, they are behaves like normal md links, not file links

**To Reproduce**
Link a file like: `[[book.pdf]]` does not work
Link a file like: `[[book.pdf| Book Volume 1]]` does not work
Link a file like: `![[book.pdf| Book Volume 1]]` does not work

**Expected behavior**
3 of them needed to behave like: `![[book.df]]`
`[Book Volume](book.pdf)` does work perfectly

**Desktop (please complete the following information):**
 - OS: Linux 
 - Browser Firefox
 - Version 108.0.2

**Comment by secure-77** [comment]
Hi @Diafwl,

yes currently file links need to start with an `!` so the only working way àt the moment is `![[book.df]]`. This is the default behavior if you drag & drop a file into a md document in obsidian.

I will take a look if I can implement the other formats as well.

**Comment by Diafwl** [comment]
but because of regex it does not capturing `![[book.pdf| Book Volume 1]]` even if this one has `!`  renaming with `|` broking file links.

**Comment by secure-77** [comment]
yes, aliases for files are not supported / implemented at the moment

**Comment by Diafwl** [comment]
if we change pdf regex to `\!?(\[\[)(.*?.pdf)(\]\])' it will work with `!` or without `!`.
i tested i little bit it seems does not broke anything.

**Comment by secure-77** [comment]
looks good 👍  then only the alias part is missing, ideally I should move the obsidian part into the PerliteParsdown.php but this is a little bit more work

**Comment by secure-77** [comment]
@Diafwl 
This is a quick and dirty solution you can try.  Just replace the pdf links section with these two:

```php
	// pdf links with Alias
	$replaces = '<a class="internal-link" target="_blank" rel="noopener noreferrer" href="'.$path .'/'.'\\2">\\3</a>';
	$pattern = array('/(\!?\[\[)(.*?.pdf)\|(.*)(\]\])/');
	$content = preg_replace($pattern, $replaces ,$content);
	
	// pdf links without Alias
	$replaces = '<a class="internal-link" target="_blank" rel="noopener noreferrer" href="'.$path .'/'.'\\2">\\2</a>';
	$pattern = array('/(\!?\[\[)(.*?.pdf)(\]\])/');
	$content = preg_replace($pattern, $replaces ,$content);
```

I will check this again as soon I have some time and will make a new release

**Comment by Diafwl** [comment]
Yeah it works perfectly.

Thank you.

**Comment by secure-77** [comment]
This is fixed in version 1.5.3
